### ISSUE TITLE
Trying to replicate the srsRAN gNB with COTS UEs Tutorial

### ISSUE THREAD

**Issue by panitsasi** [srsran/srsRAN_Project#886]
## Issue Description ##
Hi all, I am trying to replicate the srsRAN gNB with COTS UEs Tutorial but I am having several issues. First of all, these are the specs of the PC that I use to replicate the tutorial: 24-Core 4.20 GHz AMD Threadripper Pro, 128 GB  RAM(2 x 64 GB 4800 MHz DDR5 ECC/REG), Ubuntu 22.04 OS. Regarding the gNB configuration, I used exactly the same parameters as you specified, using USRP B210 and OpenPlus 8T TMO.
![image](https://github.com/user-attachments/assets/cc769d9b-ed53-4a9d-8645-e4851d18ccf9). 

For the SIM programming, i use the Gemalto IDBridge CT30 USB Smart Card Reader Writer and as programmable SIM cards I use SysmoISIM-SJA2 . For, instance i have one SIM card with the following details: 
![image](https://github.com/user-attachments/assets/1a1b243b-0aab-4f35-8677-3f2a187b32ad)

To program the SIM card, I used the following instruction: 
./pySim-prog.py -p0 -s <ICCID> --mcc=<MCC> --mnc=<MNC> -a <ADM-KEY> --imsi=<IMSI> -k <KI> --opc=<OPC>
./pySim-prog.py -p0 -s 8949440000001288206 --mcc=001 --mnc=01 -a 27044531 --imsi=999700000128820 -k 22096A80F5E0A80387EB18C905450A07 --opc=771E373D290E3DFC9018978771BAB20E
![Screenshot from 2024-10-21 13-26-46](https://github.com/user-attachments/assets/39def378-a985-43bc-a2ad-588771701e15)

Finally, when i re-read the SIM card, I did not see the services 125, 126. (read_SIM.txt attached)
[read_SIM.txt](https://github.com/user-attachments/files/17465834/read_SIM.txt)

Is there any mistake here ? I did not configure SUCI, because the services were not enabled. 

Also, I added the UE information in open5GS subscriber database. 

![image](https://github.com/user-attachments/assets/e91e90e4-093d-420e-ad3d-8228db4dd6f5)
![image](https://github.com/user-attachments/assets/85844e30-ec30-4693-8c7e-ae620f18d0cc)

I used as subscriber key :  KI, opc: OPC, AMF: 8000 (i am not sure about this), DNN/APN*: srsran, Type*: ipv4
Is everything looks okay with the previous configuration ? 

I am not sure, because when I try to create a APN in my OnePLUS 8T TMO, I see that the MCC = 999 and MNC = 70 which it does not make sense. I am trying to change these values to 001 and 01 respectively but the APN cannot be created. 
![image](https://github.com/user-attachments/assets/54644fc5-8fd4-487a-baaa-1d41a1dad209)

Also, I am facing issues when I am trying to enable 5G for the UE. 
![image](https://github.com/user-attachments/assets/45ae3c11-dd56-4114-86df-fd77b7d5ab3d)
![image](https://github.com/user-attachments/assets/8a479ae1-d67d-4449-b9ee-37edd0c547e7)
Does anyone had the same issues? 

I tried to replace the OnePlus 8T with a Google Pixel 7 but i had the same issues except the 5G connectivity. 
![image](https://github.com/user-attachments/assets/194b1b43-8de1-4c47-84f0-e61575c3d5fe)
![image](https://github.com/user-attachments/assets/545e0e7e-a3da-4794-ba9c-dc405e17b297)

For the core network, I used these conf_files for amf and nrf. 
[amf_configuration.txt](https://github.com/user-attachments/files/17465896/amf_configuration.txt)
[nrf_configuration.txt](https://github.com/user-attachments/files/17465899/nrf_configuration.txt)

Everything is running locally, so i did not change the upf.yaml file. 

The core network is running properly, and the gnb can connect with the core. 
![image](https://github.com/user-attachments/assets/a70e79ef-4f4e-4f62-8f33-ccdfae6cf271)
![image](https://github.com/user-attachments/assets/a62303f1-d684-4afc-a0ce-a7936f296c88)

Unfortunately, the UE cannot connect to the network, probably due to SIM/APN misconfiguration. Any hints would be much appreciated. 
Thank you.

**Comment by fllay** [comment]
You configuration file has plmn = 90170 but your SIM card is 99970

999700000128820

**Comment by panitsasi** [comment]
Yes, that's true. I modified it but it did not work. I am trying to replicate the roaming scenario that's why the UE PLMN is different from the core PLMN. 

gNB configuration file
![image](https://github.com/user-attachments/assets/4b1119c3-22a2-4c4c-80cf-284c46293ca0)

AMF configuration file
![image](https://github.com/user-attachments/assets/23aade5d-64b9-4acd-ba83-bd9e987d360a)

NRF configuration file 
![image](https://github.com/user-attachments/assets/d6791cc4-bab1-4a0d-afe7-69dee67dc4b0)

gNB is connecting to Core Network
![image](https://github.com/user-attachments/assets/9a066209-b281-417e-83ee-f5d7456589be)
![image](https://github.com/user-attachments/assets/8e51c8f9-a80b-4910-8dfd-89a0d78504ae)

SIM configuration after reprogramming it 
![image](https://github.com/user-attachments/assets/3db2e7f2-9e49-4440-b98d-57ac036966c2)
![image](https://github.com/user-attachments/assets/68a28a2c-94a0-4922-9b16-9ac0b184eaa6)

Updated the Core database with the SIM information
![image](https://github.com/user-attachments/assets/2abc89a2-bd50-4b89-b031-ae37a66f2d4e)
![image](https://github.com/user-attachments/assets/00ac13f2-ad95-4e9f-9d6a-347c09b7881f)

Enable the SIM, roaming and create APN
![image](https://github.com/user-attachments/assets/3691f9e8-fbd5-49c1-a2ae-63918b02ef05)
![image](https://github.com/user-attachments/assets/d882bae4-fbd7-4cfd-8566-37c59589856c)

When i tried to search for the network, the UE  did not find it. 
Does anyone faced similar issue ? 
Thanks.

**Comment by antonioo420** [comment]
Hi,

The PLMN configured in the SIM is 00101, but the PLMN on the gNB conf file, AMF and NRF is 90170. Try changing the PLMN of the SIM to 90170 and register it in the suscriber data of open5gs. Also try using the force 5G NR app that is shown in the tutorial.

**Comment by Sonam-22** [comment]
Following, I am also facing similar issues. Any help will be appreciated.

**Comment by Sonam-22** [comment]
After changing the gNB plmn and UE plmn to 00101, everything worked fine.

**Comment by pgawlowicz** [comment]
@panitsasi please check this page: https://docs.srsran.com/projects/project/en/latest/knowledge_base/source/cots_ues/source/index.html

There are some notes for OnePlus 8t and Pixel.
